builder/azure/arm/step_get_source_image_name.go

```diff
@@ -76,13 +76,13 @@ func (s *StepGetSourceImageName) Run(ctx context.Context, state multistep.StateB
 				s.say(fmt.Sprintf(" -> SourceImageName: '%s'", parentSourceID))
 				s.GeneratedData.Put("SourceImageName", parentSourceID)
 				return multistep.ActionContinue
-			} else {
-				// If the Gallery Image Version was not sourced from a Managed Image, that means it was captured directly from a VM, so we just use the gallery ID itself as the source image
-				s.say(fmt.Sprintf(" -> SourceImageName: '%s'", *image.Id))
-				s.GeneratedData.Put("SourceImageName", *image.Id)
-				return multistep.ActionContinue
 			}
-
+		}
```

Comment:
The main change here is to move block after the `if image.Properties != nil &&
			image.Properties.StorageProfile.Source != nil && image.Properties.StorageProfile.Source.Id != nil` check, since these were null on the impacted users API responses (although never in mine), since if the storage profile isn't set, we can just take the gallery image ID and be reasonable sure that's correct.